### DISCUSSION TITLE
Hoverslam mode

### DISCUSSION THREAD

**Discussion by lukeify** [ehmorris/lunar-lander#18] [Ideas]
Awesome game 👍.

Love the simplicity of it, so I can imagine you might not see this feature as a good fit. In real life, rockets have a finite number of engine firings they can perform before needing some level of refurbishment. It's also more economical in terms of fuel cost, the less time your engine is firing, the less time you're [fighting against gravity](https://en.wikipedia.org/wiki/Gravity_loss).

With some practice, I was able to perform a "hoverslam" (landing with only one main engine firing, in close proximity to the ground) twelve times out of fifty. 

Having this as a game constraint would be pretty awesome. You could calculate backwards and give the user a red/yellow/green alert when they're in the right altitude-velocity space to perform a hoverslam too, to give users some guidance.

**Comment by ehmorris** [comment]
Thanks for the idea! I have been thinking about adding a hard mode or a daily challenge, and this could be a fun addition.

**Comment by ericsampson** [comment]
adding things like this to the score algo would be very fun (different challenges or modes etc)
- main rocket fire count
- side rocket fire count
- fuel burnt
- time-to-land
